### PR TITLE
Add .clang-format to preserve_paths

### PR DESCRIPTION
**Pull request by EricMuller22**
Pod improvements

cc @alanf @dfed @dnkoutso @justinseanmartin

### CODE REVIEW

File: SpaceCommander.podspec [hunk 1]
```diff
@@ -14,5 +14,8 @@ Pod::Spec.new do |s|
   # Use this file to overcome CocoaPods limitation that at least one source file must be specified.
   s.source_files = "CocoaPodsDummy.m"
 
-  s.preserve_paths = "**/*"
+  s.preserve_paths = "**/*", ".clang-format"
+  
+  # Run setup-repo.sh when the pod is installed.
+  s.prepare_command = "./setup-repo.sh"
```
Comment (dfed) [inline SpaceCommander.podspec]: TIL!

**Comment by dfed** [comment]
GORM. We're raising the bar.

raise the bar

**Comment by dnkoutso** [comment]
LGTM.

File: SpaceCommander.podspec (hunk 1 above)
Comment (alanf) [inline SpaceCommander.podspec]: this is cool! Which directory is the command run from?

Comment (alanf) [inline SpaceCommander.podspec]: Also, if this works, we should update the README to note that the install will happen automatically if Cocoapods are used

And we should add cocoapod installation instructions there too (the things described here can be a separate PR)

Comment (dnkoutso) [inline SpaceCommander.podspec]: It runs straight from the directory the pod was installed so its relative to the path!

**Comment by EricMuller22** [comment]
We'll have to revisit prepare_command - right now that setup_repo shell script expects to be in a different directory.

**Comment by alanf** [comment]
LGTM

**Comment by EricMuller22** [comment]
@dfed @alanf - we should still look into `prepare_command` at some point, but you-know-what is successfully consuming SpaceCommander as a Pod now, so we can probably push a public version.
groundhog
